app/src/app/sections/lia/lia-submission-page/lia-submission-page.component.ts

```diff
@@ -14,31 +16,126 @@ import { LiaHeaderComponent } from './../lia-header';
   directives: [
     LiaDescriptionComponent,
     LiaCodeSubmissionComponent,
-    LiaHeaderComponent
+    LiaHeaderComponent,
+    LiaToastComponent
   ]
 })
 
 export class LiaSubmissionPageComponent implements OnInit {
 
   lia: Lia;
+  saveProgressIntervalId: number;
+  timeWarningIntervalId: number;
+  timeWarningDisplayed: boolean;
+  toastMsg: string;
 
-  constructor(private liaService: LiaService,
-              private router: Router,
-              private routeParams: RouteParams) { }
+  constructor(
+    private liaService: LiaService,
+    private router: Router,
+    private routeParams: RouteParams) {
+      this.timeWarningDisplayed = false;
+      this.toastMsg = '';
+  }
 
   ngOnInit() {
```

Comment:
`ngOnInit()` got pretty big, there should be a better separation of code according to functional pieces